Content/Items/UndergroundTemple/TempleLensSystem.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.Xna.Framework.Graphics.PackedVector;
+using StarlightRiver.Content.Items.BaseTypes;
+using StarlightRiver.Content.Items.Breacher;
+using System.Linq;
+using Terraria.ID;
+using Terraria.Graphics.Effects;
+using static Terraria.ModLoader.ModContent;
+
+namespace StarlightRiver.Content.Items.UndergroundTemple
+{
+	public class TempleLensSystem : IOrderedLoadable
+	{
+		public static RenderTarget2D NPCTarget;
+		public static RenderTarget2D LensTarget;
+
+		public float Priority => 1.1f;
+
+		public void Load()
+		{
+			if (Main.dedServ)
+				return;
+
+			ResizeTarget();
+
+			On.Terraria.Main.DrawNPCs += DrawLens;
+			On.Terraria.Main.CheckMonoliths += Main_CheckMonoliths;
+		}
+
+		public void Unload()
+		{
+			if (Main.dedServ)
+				return;
+
+			On.Terraria.Main.DrawNPCs -= DrawLens;
+			On.Terraria.Main.CheckMonoliths -= Main_CheckMonoliths;
+		}
+
+		private void Main_CheckMonoliths(On.Terraria.Main.orig_CheckMonoliths orig)
+		{
+			GraphicsDevice gD = Main.graphics.GraphicsDevice;
+			SpriteBatch spriteBatch = Main.spriteBatch;
+
+			if (Main.gameMenu || Main.dedServ || spriteBatch is null || NPCTarget is null || gD is null || (!Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Buffs.Illuminant>()))) && !Main.npc.Any(n => n.active && n.HasBuff(ModContent.BuffType<Exposed>())))
+				return;
+
+			#region npctarget
```

Comment:
Is there a reason for a region thats a grand total of 4 lines shorter than the entire method? Seems a bit redundant and pointlessly confusing

Reply:
Fixed

Reply:
Lets replace these regions with comments -- they're entirely contained within one method making the use case of collapsing them not very useful